chore: Add support for remote-repo-url flag [CFG-2085]

ofekatr opened the pull request:
### What this does

Add support for `remote-repo-url` flag in the new engine. This flag will override the remote repository URL when sharing results via `--report --remote-repo-url=<new-url>`.

If the `target-name` flag is used too, then the `--target-name` will override that for the project name.

PR https://github.com/snyk/snyk-iac-test/pull/89 will process the `remote-repo-url` flag in `snyk-iac-test`. 

### Notes for the reviewer

To test: 
- Build `snyk-iac-test` from https://github.com/snyk/snyk-iac-test/pull/89:
```
go build -o snyk-iac-test .
```
- run `SNYK_IAC_POLICY_ENGINE_PATH=<path-to-snyk-iac-test>  snyk-dev iac test test/fixtures/iac/terraform/sg_open_ssh.tf --experimental --report --remote-repo-url=yolo`
- verify the project was created under the name `yolo`.
- run `SNYK_IAC_POLICY_ENGINE_PATH=<path-to-snyk-iac-test>  snyk-dev iac test test/fixtures/iac/terraform/sg_open_ssh.tf --experimental --report --remote-repo-url=oloy --target-name=yolo`
- verify the project was created under the name `yolo`.

### More information

[CFG-2085](https://snyksec.atlassian.net/browse/CFG-2085)

ipapast commented:
this is not going to work with the current snyk-iac-test version. could you make sure to update the checksums again after this PR is merged? https://github.com/snyk/snyk-iac-test/pull/90

ofekatr commented:
> this is not going to work with the current snyk-iac-test version. could you make sure to update the checksums again after this PR is merged? https://github.com/snyk/snyk-iac-test/pull/90 

Forgot to push after I changed the flag name, thanks for catching this 👍 
I've updated the checksum values to be the ones for `v0.21.1`
